File: functions/ffmpeg/variants/go/3_weight_2.go
package main

func weight(i int32, blen int32, offset int32) int32 {
    if i < 2*offset {
        if offset == 1 {
            if i != 0 {
                return 5
            }
            return 3
        }
        return 1 + (6*i + offset - 1) / (2*offset - 1)
    }
    if i > blen-1-2*offset {
        if offset == 1 {
            if blen-1-i != 0 {
                return 5
            }
            return 3
        }
        return 1 + (6*(blen-1-i) + offset - 1) / (2*offset - 1)
    }
    return 8
}


func main(){
    weight(0,0,0)
}

File: functions/ffmpeg/variants/c/3_weight_2.c
#include <stdint.h>

static int weight(int i, int blen, int offset)
{
    int result;
    if (i < 2 * offset) {
        result = offset == 1 ? (i ? 5 : 3) : (1 + (6 * i + offset - 1) / (2 * offset - 1));
    } else if (i > blen - 1 - 2 * offset) {
        result = offset == 1 ? ((blen - 1 - i) ? 5 : 3) : (1 + (6 * (blen - 1 - i) + offset - 1) / (2 * offset - 1));
    } else {
        result = 8;
    }
    return result;
}


int main(){
    weight(0,0,0);
}
